tests/impl/async_end_transaction.cpp

```diff
@@ -0,0 +1,70 @@
+#include "connection_mock.h"
+
+#include <ozo/impl/async_end_transaction.h>
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+namespace {
+
+using namespace testing;
+using namespace ozo::tests;
+
+using ozo::error_code;
+
+struct async_end_transaction : Test {
+    StrictMock<connection_gmock> connection {};
+    StrictMock<executor_gmock> callback_executor{};
+    StrictMock<callback_gmock<connection_ptr<>>> callback {};
+    StrictMock<executor_gmock> executor {};
+    StrictMock<executor_gmock> strand {};
+    StrictMock<strand_executor_service_gmock> strand_service {};
+    StrictMock<stream_descriptor_gmock> socket {};
+    StrictMock<steady_timer_gmock> timer {};
+    io_context io {executor, strand_service};
+    decltype(make_connection(connection, io, socket, timer)) conn = make_connection(connection, io, socket, timer);
+};
+
+TEST_F(async_end_transaction, should_call_async_execute) {
+    *conn->handle_ = native_handle::good;
+
+    auto transaction = ozo::impl::transaction<decltype(conn)>(std::move(conn));
+
+    Sequence s;
+
+    EXPECT_CALL(executor, dispatch(_)).InSequence(s).WillOnce(InvokeArgument<0>());
+    EXPECT_CALL(strand_service, get_executor()).InSequence(s).WillOnce(ReturnRef(strand));
+
+    // Send query params
+    EXPECT_CALL(connection, set_nonblocking()).InSequence(s).WillOnce(Return(0));
```

Comment:
This is very low-level details which are not the test subject. Here must be mocked async_execute() call and no more.